frame/support/test/tests/pallet_ui/storage_info_unsatisfied.stderr

```diff
@@ -1,8 +1,8 @@
 error[E0277]: the trait bound `Bar: MaxEncodedLen` is not satisfied
-  --> $DIR/storage_info_unsatisfied.rs:10:12
+  --> $DIR/storage_info_unsatisfied.rs:23:16
    |
-10 |     #[pallet::generate_storage_info]
-   |               ^^^^^^^^^^^^^^^^^^^^^ the trait `MaxEncodedLen` is not implemented for `Bar`
+23 |     type Foo<T> = StorageValue<_, Bar>;
+   |                   ^^^^^^^^^^^^ the trait `MaxEncodedLen` is not implemented for `Bar`
```

Comment:
I think the span that makes the most sense to me is pointing directly at the type where the error message is complaining about:

```suggestion
23 |     type Foo<T> = StorageValue<_, Bar>;
   |                                   ^^^ the trait `MaxEncodedLen` is not implemented for `Bar`
```

but it feels like you opted for the storage type instead because it's impossible to determine which inner type the compiler is complaining about? If so, then perhaps pointing to the angle brackets would make more sense?

```suggestion
23 |     type Foo<T> = StorageValue<_, Bar>;
   |                               ^^^^^^^^ the trait `MaxEncodedLen` is not implemented for `Bar`
```

Reply:
I tried to put the span on the whole type, I don't know yet why it only covers the ident `StorageValue`.
I'm not sure we can point to a particular generic because other generic can potentially be wrong.

But I'm thinking the best error message is for a failing `#[pallet::getter(fn ...)]`

it looks like this:
```
error[E0277]: the trait bound `Foo: WrapperTypeDecode` is not satisfied
   --> frame/support/test/tests/pallet.rs:231:36
    |
231 |     pub type Bar<T> = StorageValue<_, Foo>;
    |                                       ^^^ the trait `WrapperTypeDecode` is not implemented for `Foo`
    |
    = note: required because of the requirements on the impl of `_::_parity_scale_codec::Decode` for `Foo`
    = note: required because of the requirements on the impl of `FullCodec` for `Foo`
    = note: required by `hidden_include::StorageValue::get`

```

If we can create dead code which triggers those error when the generic doesn't implement the wanted trait (so Value should implement FullCodec, Key also, Hasher etc...) then I think that would be the ideal.

Reply:
after thought, we can just write a code very similar to the code generated for the getter to get the wanted error message. Shouldn't be difficult.

Reply:
I just realize that getter has better span because in the code we write: `<storagealias<T> as StorageValue<Value>>::get` so the error points to `Value` and it got the correct spans.

Reply:
> but it feels like you opted for the storage type instead because it's impossible to determine which inner type the compiler is complaining about? If so, then perhaps pointing to the angle brackets would make more sense?

Yes exactly, that said on my nightly it uses the span of the whole type with its generics. but stable only points to the ident